lib/core/config.js

```diff
@@ -231,30 +231,26 @@ Config.prototype.loadContractsConfigFile = function() {
   });
 
   let configFilePath = this._getFileOrOject(this.configDir, 'contracts', 'contracts');
-
+  const autoGas = "auto";
   let newContractsConfig = this._mergeConfig(configFilePath, configObject, this.env);
-  if (newContractsConfig.gas != "auto") {
-      newContractsConfig.gas = utils.getWeiBalanceFromString(newContractsConfig.gas, web3);
+  if (newContractsConfig.gas != autoGas) {
+    newContractsConfig.gas = utils.getWeiBalanceFromString(newContractsConfig.gas, web3);
   }
-  if ('deployment' in newContractsConfig) {
-    if ('accounts' in newContractsConfig.deployment) {
-      for (var i = 0; i < newContractsConfig.deployment.accounts.length; i++) {
-        newContractsConfig.deployment.accounts[i].balance = utils.getWeiBalanceFromString(newContractsConfig.deployment.accounts[i].balance, web3);
-      }
-    }
+  if ('accounts' in newContractsConfig.deployment) {
+    newContractsConfig.deployment.accounts.forEach((account, index) => {
+      account.balance = utils.getWeiBalanceFromString(account.balance, web3);
+    });
   }
-  for (var contractName in newContractsConfig.contracts) {
-    if (newContractsConfig.contracts[contractName].gas != "auto") {
-      if (newContractsConfig.contracts[contractName].gas) {
-        newContractsConfig.contracts[contractName].gas = utils.getWeiBalanceFromString(newContractsConfig.contracts[contractName].gas, web3);
+  Object.keys(newContractsConfig.contracts).forEach(contractName => {
+      let contractGas = newContractsConfig.contracts[contractName].gas;
+      let contractGasPrice = newContractsConfig.contracts[contractName].gasPrice;
+      if (contractGas && contractGas != autoGas) {
```

Comment:
So in JavaScript there's this thing with loose equality vs strict equality. For example:

```
'0' == 0 // true
'0' === 0 // false
```

Therefore, it's generally "best practice" to always go with strict equality, unless the use case doesn't allow it. 

So maybe we can change `!=` to `!==` here.